### ISSUE TITLE
Render pplastic material using mesh vertex attributes stored in a .PLY 3D model

### ISSUE THREAD

**Issue by ZhaoJinyu96** [mitsuba-renderer/mitsuba2#496]
- [❔ other question] 

## Summary

I can use a texture to set spatially varying diffuse albedos of 'pplastic' material, but I wonder whether the diffuse albedo can be assigned spatially using the color of vertices stored in .ply 3D models.

## System configuration

- Platform: Windows
- Compiled variants:
    * `scalar_rgb`
    * `scalar_spectral`
    * `scalar_rgb_polarized`
    * `scalar_mono_polarized`
    * `scalar_spectral_polarized`

## Description

I used your sample code of 'bsdf_pplastic.xml', and changed the uniform albedo to a texture.
I also tried to set the albedo according to the vertex color stored in the .ply model but it says that 
unreferenced object MeshAttribute[
[xml.cpp:1059]   name = "vertex_color",
[xml.cpp:1059]   scale = "1"
[xml.cpp:1059] ] (within bsdf of type "pplastic")
Is there something wrong or 'pplastic' does not support MeshAttribute plugin?

[bsdf_pplastic.zip](https://github.com/mitsuba-renderer/mitsuba2/files/7143482/bsdf_pplastic.zip)

**Comment by tizian** [comment]
Could you please post the full version of your test scene that also includes your `matpreview.xml` file and the `.ply` mesh?

**Comment by ZhaoJinyu96** [comment]
This is the full version of my test scene (I modified docs/scenes/bsdf_pplastic.xml).
[sample.zip](https://github.com/mitsuba-renderer/mitsuba2/files/7143634/sample.zip)

I had no problem when I used diffuse BSDF.

**Comment by tizian** [comment]
Ah ok, I see what is happening now. It's a combination of two things:

- The diffuse color argument of the `pplastic` plugin is actually called `diffuse_reflectance` (instead of only `reflectance` in the `diffuse` BSDF)
- The `mesh_attribute` texture does not implement the `mean` function that is used during importance sampling of the `pplastic`. To work around that, you can go into `pplastic.cpp` and adjust the `parameters_changed` function:
```
void parameters_changed(const std::vector<std::string> &/*keys*/ = {}) override {
    m_specular_sampling_weight = 0.5f;
}
```

**Comment by ZhaoJinyu96** [comment]
Thank you very much. Now the problem is solved. 
I have another question about the format of vertex color. As far as I see, most .ply models store vertex color as "uchar" instead of "float" as the sample mesh, so do I have to change the "uchar" color [0,255] to "float" [0,1] first?

**Comment by ZhaoJinyu96** [comment]
Thank you very much! The problem is solved.
I have another question about the format of vertex color stored in .ply models.
As far as I see, vertex color is stored as "uchar" rather than "float" as the sample mesh. 
When I used other .ply models, it says that 
"Mesh attribute "vertex_color" has integer fields: color attributes are expected to be in the [0, 1] range."
Tools I used seem to support "uchar" only and I wonder if there is any tool to convert "uchar" color [0,255] to "float" color [0,1]?
